lobster/io.py

```diff
@@ -124,10 +125,18 @@ def lobster_read(mh, filename, level, items, source_info=None):
 
         if all(filter_conditions):
             if item.tag.key() in items:
-                mh.error(item.location,
-                          "duplicate definition of %s, "
-                          "previously defined at %s" %
-                          (item.tag.key(),
-                           items[item.tag.key()].location.to_string()))
-
-            items[item.tag.key()] = item
+                # 'duplicate definition' errors are fatal, but the user wants to see all
+                # of them. So store the affected items in a list first, and create
+                # errors later.
+                duplicate_items.append(item)
+            else:
+                items[item.tag.key()] = item
+
+    if duplicate_items:
+        for i, item in enumerate(duplicate_items, start=1):
```

Comment:
Suggestion - Instead of using 'i' can a meaningful name be added here?

Reply:
Done.